lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -585,16 +585,15 @@ def from_relation_data(cls, data: dict):
     def identifier(self) -> str:
         """Format the topology information into a terse string."""
         # This is odd, but may have `None` as a model key
-        return "_".join([str(val) for val in self.as_dict().values()]).replace("/", "_")
+        return "_".join([str(val) for val in self.as_promql_label_dict().values()]).replace(
+            "/", "_"
+        )
 
     @property
     def promql_labels(self) -> str:
         """Format the topology information into a verbose string."""
         return ", ".join(
-            [
-                'juju_{}="{}"'.format(key, value)
-                for key, value in self.as_dict(rename_keys={"charm_name": "charm"}).items()
-            ]
+            ['{}="{}"'.format(key, value) for key, value in self.as_promql_label_dict().items()]
```

Comment:
This looks significantly different. Is this a breaking change?

Reply:
It's the same except without unit. `as_promql_label_dict` applies `"juju_" to the values.

Reply:
I am resolving this discussion so we can move conversation to https://github.com/canonical/prometheus-k8s-operator/issues/240